q2cli/util.py

```diff
@@ -178,3 +182,26 @@ def callback(ctx, param, value):
     return click.Option(['--citations'], is_flag=True, expose_value=False,
                         is_eager=True, callback=callback,
                         help='Show citations and exit.')
+
+
+def usage_example_option(action):
+    import click
+    import q2cli.core.cache
+
+    def callback(ctx, param, value):
+        if not value or ctx.resilient_parsing:
+            return
+
+        action = ctx.command._get_action()
+        plugin_id = to_cli_name(action.plugin_id)
+        cached_plugin = q2cli.core.cache.CACHE.plugins[plugin_id]
+        cached_action = cached_plugin['actions'][action.id]
+        msg = cached_action.get("examples")
+        if not msg:
+            msg = "No examples have been registered for this action yet."
+        click.secho(msg)
+        ctx.exit()
```

Comment:
@thermokarst is this sufficient for leveraging rendered usage examples that have been cached?

Reply:
I think so! I'll take a closer look soon, but this looks reasonable to me. Thanks!